src/utils/trpc.ts

```diff
@@ -9,13 +9,38 @@ import { TRPCWithErrorCodeSchema } from '~/utils/error'
 import type { AppRouter } from '~/server/modules/_app'
 import { getBaseUrl } from './getBaseUrl'
 import { type TRPC_ERROR_CODE_KEY } from '@trpc/server/rpc'
+import { LOGGED_IN_KEY } from '~/constants/insecureCookies'
+import Router from 'next/router'
+import { SIGN_IN } from '~/lib/routes'
+import { CALLBACK_URL_KEY } from '~/constants/params'
+import { deleteCookie } from 'cookies-next'
 
 const NON_RETRYABLE_ERROR_CODES: Set<TRPC_ERROR_CODE_KEY> = new Set([
   'UNAUTHORIZED',
   'FORBIDDEN',
   'NOT_FOUND',
 ])
 
+const handleErrorsOnClient = (error: unknown): boolean => {
+  if (typeof window === 'undefined') return false
+  if (!(error instanceof TRPCClientError)) return false
+
+  if (error.data?.code === 'UNAUTHORIZED') {
```

Comment:
nit: why dont we put the logic of cookie deletion and routing/redirecting in  src/pages/api/trpc/[trpc].ts `onError` instead (which is the current behaviour of the secure cookie now)?

this way we dont have to put it in the retry section of queryClientConfig since this logic is not really about retrying of the queries in trpc. :)

Reply:
Good point, will shift! That makes sense.

Reply:
Ah, I remember why the redirect is in here, cause the [trpc] api route only invokes on the server and cannot redirect. We can think of this file as client side handling (which includes the deletion of a client-side-readable cookie and the redirection rules). Will add a comment in the code.